### DISCUSSION TITLE
MediaPipe Pose Detection

### DISCUSSION THREAD

**Discussion by UUoocl** [WarmUpTill/SceneSwitcher#1037] [Show and tell]
I have a project where the idea is to use hand and body positions to control OBS.

Google [MediaPipe Pose Landmarks](https://developers.google.com/mediapipe/solutions/vision/pose_landmarker) are sent to OBS and trigger Advanced Scene Switcher conditions. 

https://github.com/UUoocl/MediaPipe_for_OBS

[Electron](https://www.electronjs.org/) is used to run the MediaPipe model locally. 

  ```mermaid
graph TD;
subgraph OBS
A(Sources)-->B(Video Projector Window);
A<-->D[Advanced Scene Switcher];
end

B(Projector Window)--> C(MediaPipe for OBS);

C <---> |WebSocket|A;
```

The body landmarks are stored in an OBS Text Source as JSON.

<img width="300" alt="pose_landmarks_index" src="https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/5df56710-92ae-4713-999f-4e0f1faa1d11">

I'd appreciate any feedback on this project.

**Comment by WarmUpTill** [comment]
Seems like a cool idea! :)

Just one question:
What exactly is the purpose of the text source?
I am not familiar if / how text sources can be used by Browser sources, so I apologize if the answer is obvious.

If it is only intended to be a proxy to receive the landmark information from and pass it into advanced scene switcher variables, you might want to consider sending that information directly via [websocket messages to the advanced scene switcher](https://github.com/WarmUpTill/SceneSwitcher/wiki/Websockets#sending-messages-to-the-scene-switcher-from-external-tools-via-websocket-requests).

So something like this:
![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/6c145fcc-9794-406e-8ef9-9f27b3e5c2f7)

**Comment by UUoocl** [comment]
Thanks for commenting.  I wasn't sure how to get the landmark information directly into Advanced Scene Switcher.  That is a helpful tip.

The Text Source is just being used to store the landmark information.  The Text Source is being read by both Advanced Scene Switcher and a Browser Source with a local HTML/javascript file.       

The page in the Browser source is using the [OBS WebSocket Js](https://github.com/obs-websocket-community-projects/obs-websocket-js) to read the content of the Text Source.  The java Script in the browser reacts to the landmark information.
